src/Chapter3.hs

```diff
@@ -480,6 +511,45 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data Castle = MkCastle String deriving (Show)
+data Facility = Church | Library deriving (Show)
+data Inhabitants = One | Two | Three | Four deriving (Show)
+data House = MkHouse Inhabitants deriving (Show)
+
+inhabitantsPerHouse :: House -> Int
+inhabitantsPerHouse (MkHouse One) = 1
+inhabitantsPerHouse (MkHouse Two) = 2
+inhabitantsPerHouse (MkHouse Three) = 3
+inhabitantsPerHouse (MkHouse Four) = 4
```

Comment:
To reduce some code duplication, you can use the `case-of` expression in such case:

```haskell
inhabitantsPerHouse (MkHouse h) = case h of
    One -> 1
    ... -> ...
```